### PR TITLE
[AMQ-7502] Remove leveldb

### PR DESCRIPTION
**Pull request by jbonofre**

### CODE REVIEW

**Comment by michaelandrepearce** [comment]
I get removal of leveldb not sure why removal of broker partitioning has been lumped in at the same time

**Comment by jbonofre** [comment]
@michaelandrepearce both are "related", as activemq-partition uses activemq-leveldb zookeeper client. Do you think it makes sense to keep partition ?

**Comment by michaelandrepearce** [comment]
Very much so. Partition is something given time something id like to port to artemis, its something when doing so would make it more an api so its not coupled as tightly with zk (e.g. zk just would be a pluggable option) but it certainly is very nice feature that helps broker setup be scaleable without manually having to shard/partition flows as a user

**Comment by jbonofre** [comment]
@michaelandrepearce ok thanks. I'm refactoring activemq-partition to avoid the dependency to leveldb and I will let there.

**Comment by lucastetreault** [comment]
Can one of the admins verify this patch?

**Comment by jbonofre** [comment]
Rebasing and keeping partition.

**Comment by jbonofre** [comment]
@michaelandrepearce I have updated the PR to keep `activemq-partition` and remove `activemq-leveldb`.

**Comment by jbonofre** [comment]
I'm fixing the compilation issue due to rebase. I will update the PR soon.

**Comment by michaelandrepearce** [comment]
Ill have a look next week on this. But sounds promising.

**Comment by jbonofre** [comment]
@michaelandrepearce did you have time to review this PR ? If you have some time to do so, it would be great. I would like to merge this PR pretty soon as it's blocking my other PRs (JDK11 build, Spring5 update, Log4j2 update, ...). Thanks !

**Comment by mattrpav** [comment]
Thanks for tackling the partition feature as part of this refactor.

**Comment by jbonofre** [comment]
I'm rebasing on `master`. As said in my previous comment: can you please verify and eventually approve this PR ? It blocks me to create the dependency PRs (JDK11 build, Spring5 update, Log4j2 update, ...).

**Review by mattrpav — COMMENTED** [review]
Looks like a few leaky changes not 100% related to leveldb removal are pulled in. Not the worst thing in the world. Maybe tag the commit w/ add'l JIRA numbers to track back?

apache-http/pom.xml
apache-shiro/pom.xml
unit-tests/.. removal of Google collections dependencies

**Review by mattrpav — APPROVED** [review]
I think we eventually should move zookeeper to activemq-partition-zookeeper to break off impl into a separate optional module, but given the chunkiness of the change its good for now.

**Comment by jbonofre** [comment]
@mattrpav it's dependent updates (removing some useless dependencies now).

File: activemq-partition/src/main/java/org/apache/activemq/partition/ZKClient.java [hunk 1]
```diff
@@ -39,30 +25,32 @@
 import org.linkedin.util.clock.Timespan;
 import org.linkedin.util.concurrent.ConcurrentUtils;
 import org.linkedin.util.io.PathUtils;
-import org.linkedin.zookeeper.client.ChrootedZKClient;
-import org.linkedin.zookeeper.client.IZooKeeper;
-import org.linkedin.zookeeper.client.IZooKeeperFactory;
-import org.linkedin.zookeeper.client.LifecycleListener;
-import org.linkedin.zookeeper.client.ZooKeeperFactory;
-import org.osgi.framework.InvalidSyntaxException;
```
Comment (michaelandrepearce) [inline activemq-partition/src/main/java/org/apache/activemq/partition/ZKClient.java:47]: why are we removing osgi support?

Comment (michaelandrepearce) [reply activemq-partition/src/main/java/org/apache/activemq/partition/ZKClient.java:47]: ignore this, just realised its exceptions that you've removed from method throws

**Review by michaelandrepearce — APPROVED** [review]

**Comment by michaelandrepearce** [comment]
> @michaelandrepearce did you have time to review this PR ? If you have some time to do so, it would be great. I would like to merge this PR pretty soon as it's blocking my other PRs (JDK11 build, Spring5 update, Log4j2 update, ...). Thanks !

Hi @jbonofre sorry just been busy with work, haven't gone through every change, but regarding the bits around partitioning which i had raised, it looks good to me.

**Comment by jbonofre** [comment]
Rebasing to fix the build issue.

**Comment by jbonofre** [comment]
@michaelandrepearce @mattrpav thanks for your review guys ! I'm rebasing/fixing build and I will merge. Thanks again !

**Pull request merged**